ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2837,264 @@ public sealed class Bot : IAsyncDisposable {
+				Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), (uint Sets, byte CardsPerSet)> itemsToTakePerInventorySet = (await Utilities.InParallel(loadCardCountsTasks).ConfigureAwait(false)).ToDictionary(entry => entry.Key, entry => entry.Value);
+				HashSet<Steam.Asset> result = GetItemsForFullBadge(inventory, itemsToTakePerInventorySet);
+
+				if (result.Count > 0) {
+					await Actions.SendInventory(result).ConfigureAwait(false);
+				}
+			} finally {
+				SendCompleteSetsSemaphore.Release();
+			}
+		}
+
+		internal static HashSet<Steam.Asset> GetItemsForFullBadge(IReadOnlyCollection<Steam.Asset> inventory, Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), (uint SetsToExtract, byte CardsPerSet)> amountsToExtract) {
+			if ((inventory == null) || (amountsToExtract == null)) {
+				throw new ArgumentNullException($"{nameof(inventory)} || {nameof(amountsToExtract)}");
+			}
+
+			if ((amountsToExtract.Count == 0) || (inventory.Count == 0)) {
+				return new HashSet<Steam.Asset>(0);
+			}
+
+			HashSet<Steam.Asset> result = new HashSet<Steam.Asset>(Convert.ToInt32(amountsToExtract.Values.Select(setInformation => setInformation.SetsToExtract * setInformation.CardsPerSet).Aggregate((a, b) => a + b)));
+			Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, List<Steam.Asset>>> itemsPerClassIDPerSet = inventory.GroupBy(item => (item.RealAppID, item.Type, item.Rarity)).ToDictionary(grouping => grouping.Key, grouping => grouping.GroupBy(item => item.ClassID).ToDictionary(group => group.Key, group => group.ToList()));
+
+			foreach (((uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity) set, (uint setsToExtract, byte cardsPerSet)) in amountsToExtract) {
+				if (!itemsPerClassIDPerSet.TryGetValue(set, out Dictionary<ulong, List<Steam.Asset>>? itemsPerClassID)) {
+					continue;
+				}
+
+				if (cardsPerSet != itemsPerClassID.Count) {
+					continue;
+				}
+
+				foreach (List<Steam.Asset> itemsOfClass in itemsPerClassID.Values) {
+					long classRemaining = setsToExtract;
+					int i = 0;
+
+					while ((classRemaining > 0) && (i < itemsOfClass.Count)) {
+						Steam.Asset item = itemsOfClass[i++];
+
+						if (classRemaining - item.Amount < 0) {
+							result.Add(
+								new Steam.Asset(
+									item.AppID,
+									item.ContextID,
+									item.ClassID,
+									Convert.ToUInt32(classRemaining),
+									item.InstanceID,
+									item.AssetID,
+									item.Marketable,
+									item.Tradable,
+									item.Tags,
+									item.RealAppID,
+									item.Type,
+									item.Rarity
+								)
```

Comment:
This is abomination, I think I added `MemberwiseClone` function to `Steam.Asset`, use it and just redefine amount.